tests/test_pyav.py

```diff
@@ -0,0 +1,329 @@
+        stream.height = frames.shape[1]
+
+        for frame in frames:
+            av_frame = av.VideoFrame.from_ndarray(frame, format="rgb24")
+            for packet in stream.encode(av_frame):
+                container.mux(packet)
+
+        for packet in stream.encode():
+            container.mux(packet)
+
+    mp4_bytes = iio.imwrite(
+        "<bytes>",
+        frames,
+        format_hint=".mp4",
+        plugin="pyav",
+        codec="libx264",
+    )
+
+    expected = Path(tmp_path / "expected.mp4").read_bytes()
+    # actual = Path(tmp_path / "actual.mp4").read_bytes()
+    assert expected == mp4_bytes
+
+
+def test_metadata(test_images: Path):
+    with iio.imopen(str(test_images / "cockatoo.mp4"), "r", plugin="pyav") as plugin:
+        meta = plugin.metadata()
+        assert meta["profile"] == "High 4:4:4 Predictive"
+        assert meta["codec"] == "h264"
+
+        plugin.metadata(index=4)
+    print("")
+
+
+def test_properties(test_images: Path):
+    with iio.imopen(str(test_images / "cockatoo.mp4"), "r", plugin="pyav") as plugin:
+        with pytest.raises(IOError):
+            # subsampled format
+            plugin.properties(format="yuv420p")
+
+        with pytest.raises(IOError):
+            # components per channel differs
+            plugin.properties(format="nv24")
+
+        props = plugin.properties()
+        assert props.shape == (3, 720, 1280)
+        assert props.dtype == np.uint8
+        assert props.is_batch is False
+
+        props = plugin.properties(index=4, format="rgb48")
+        assert props.shape == (720, 1280, 3)
+        assert props.dtype == np.uint16
+        assert props.is_batch is False
+
+        props = plugin.properties(index=None)
+        assert props.shape == (280, 3, 720, 1280)
+        assert props.dtype == np.uint8
+        assert props.is_batch is True
+
+
+def test_video_format_to_dtype():
+    fake_formats = [
+        "vdpau",
+        "dxva2_vld",
+        "vaapi_moco",
+        "drm_prime",
+        "mediacodec",
+        "qsv",
+        "mmal",
+        "vulkan",
+        "vaapi_vld",
+        "d3d11",
+        "d3d11va_vld",
+        "videotoolbox_vld",
+        "vaapi_idct",
+        "opencl",
+        "cuda",
+        "xvmc",
+    ]
+
+    for name in video_format_names:
+        format = av.VideoFormat(name)
+
+        if name in fake_formats:
+            # should fail
+            with pytest.raises(ValueError):
+                _format_to_dtype(format)
+        else:
+            # should succseed
+            _format_to_dtype(format)
+
+
+def test_filter_sequence(test_images):
+    frames = iio.imread(
+        test_images / "cockatoo.mp4",
+        index=None,
+        plugin="pyav",
+        filter_sequence=[
+            ("drawgrid", "w=iw/3:h=ih/3:t=2:c=white@0.5"),
+            ("scale", {"size": "vga", "flags": "lanczos"}),
+            ("tpad", "start=3"),
+        ],
+    )
+
+    assert frames.shape == (283, 3, 480, 640)
```

Comment:
```suggestion
    # cockatoo.mp4 uses yuv444p, which is a planar format 
    # and it will thus come out as time, channel, height, width. 
    assert frames.shape == (283, 3, 480, 640)

    frames = iio.imread(
        test_images / "cockatoo.mp4",
        index=None,
        plugin="pyav",
        # passing format "rgb24" ensures the images come out as 
        # time, height, width, channel
        format="rgb24",
        filter_sequence=[
            ("drawgrid", "w=iw/3:h=ih/3:t=2:c=white@0.5"),
            ("scale", {"size": "vga", "flags": "lanczos"}),
            ("tpad", "start=3"),
        ],
    )

    assert frames.shape == (283, 480, 640, 3)
```

Reply:
@hmaarrfk Done. After discussion with @almarklein and with local colleagues I changed the default for reading in pyav to `rgb24`. Passing `format=None` will still read the native colorspace, but it will no longer be the default. I've added a test to make sure both scenarios are covered.